config/types/iptables.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"errors"
+	"fmt"
+
+	ignTypes "github.com/coreos/ignition/config/v2_0/types"
+	"github.com/coreos/ignition/config/validate/report"
+	"github.com/vincent-petithory/dataurl"
+)
+
+var (
+	ErrInvalidIptablesDefault = errors.New("iptables chain default must be one of ACCEPT or DROP")
+)
+
+type Iptables struct {
+	V4 *iptablesChains `yaml:"v4"`
+	V6 *iptablesChains `yaml:"v6"`
+}
+
+type iptablesChains struct {
+	Input   iptablesRules `yaml:"input"`
+	Forward iptablesRules `yaml:"forward"`
+	Output  iptablesRules `yaml:"output"`
+}
+
+type iptablesRules struct {
+	Default string   `yaml:"default"`
+	Rules   []string `yaml:"rules"`
+}
+
+func (ir iptablesRules) Validate() report.Report {
```

Comment:
If we give Default it's own type (aliased to string) and define Validate() on that instead, it can give errors with greater precision.